modules/database/src/adapters/utils/validateFieldConstraints.ts

```diff
@@ -1,16 +1,36 @@
 import { ConduitError, ConduitModel, ConduitModelField } from '@conduitplatform/grpc-sdk';
 import { ConduitDatabaseSchema } from '../../interfaces';
+import { isObject } from 'lodash';
 
 /*
  * Validates schema field constraints.
  * 'unique' requires 'required'
  */
-export function validateFieldConstraints(schema: ConduitDatabaseSchema) {
-  fieldsValidator(schema.name, schema.compiledFields);
+export function validateFieldConstraints(schema: ConduitDatabaseSchema, db: string) {
+  fieldsValidator(schema.name, schema.compiledFields, db);
 }
 
-export function fieldsValidator(schemaName: string, schemaFields: ConduitModel) {
+export function fieldsValidator(
+  schemaName: string,
+  schemaFields: ConduitModel,
+  db: string,
+  blockRelations = false,
+) {
   Object.keys(schemaFields).forEach(f => {
+    if (/(?<=[a-zA-Z0-9])_(?=[a-zA-Z0-9])|(?<=[a-zA-Z0-9])_$/g.test(f)) {
+      throw new ConduitError(
+        'INVALID_ARGUMENTS',
+        400,
+        `Schema '${schemaName}' violates field '${f}' constraint (field names cannot contain '_' in-between characters).`,
```

Comment:
What if we used double underscores instead?
That would preserve support for more field names (probably not uncommon for foreign dbs, also digit words), while also retaining indexOf() checks.

Reply:
Double underscores are not common practice for signifying "objects" in tables, and it'd be very ugly in practice. The only other applicable solution would be to do something smarter in wrap/unwrap ops that wouldn't confuse random underscores as the existence of an object.

Reply:
Also this is a db rework issue, and your concerns should be addressed at the other PR